rs/tests/testing_verification/testnets/from_config.rs

```diff
@@ -52,34 +107,111 @@ pub fn setup(env: TestEnv) {
     );
 
     if let Some(boundary_nodes) = parsed.boundary_nodes {
-        for bn in boundary_nodes {
-            BoundaryNode::new(bn.name)
-                .allocate_vm(&env)
-                .expect("Allocation of BoundaryNode failed.")
-                .for_ic(&env, "")
-                .use_real_certs_and_dns()
-                .start(&env)
-                .expect("Failed to setup BoundaryNode VM");
-        }
+        boundary_nodes.iter().for_each(|bn| {
+            match bn {
+                ConfigurableBoundaryNode::Simple(bn) => BoundaryNode::new(bn.name.clone()),
+                ConfigurableBoundaryNode::Complex(b) => *b.to_owned(),
+            }
+            .allocate_vm(&env)
+            .expect("Allocation of BoundaryNode failed.")
+            .for_ic(&env, "")
+            .use_real_certs_and_dns()
+            .start(&env)
+            .expect("Failed to setup BoundaryNode VM")
+        })
     }
 
     env.sync_with_prometheus();
 }
 
+fn write_file_and_update_env_variable(env: &TestEnv, pairs: Vec<(&str, String, &str)>) {
+    for (file_name, value_in_file, env_variable) in pairs {
+        let path = env.get_dependency_path(file_name);
+        std::fs::write(&path, value_in_file)
+            .unwrap_or_else(|_| panic!("Failed to write to path: {}", path.display()));
```

Comment:
What's the purpose of writing these files?

Reply:
environment values only give the path to files which contain information. That is how `HasDependencies`  trait is implemented